arch/risc-v/src/common/supervisor/riscv_perform_syscall.c

```diff
@@ -65,6 +65,10 @@ void *riscv_perform_syscall(uintreg_t *regs)
 
       addrenv_switch(NULL);
 #endif
+      /* Update scheduler parameters */
+
+      nxsched_suspend_scheduler(g_running_tasks[this_cpu()]);
```

Comment:
should  we save the result at line 40 to avoid the dup?

Reply:
I just copied this from https://github.com/apache/nuttx/blame/759b77b975780fc4a5c26414e4762e9e9d5e5f17/arch/risc-v/src/common/riscv_doirq.c#L114-L118

Reply:
should we need update both:)

Reply:
Sure, I can fix it for riscv, I'll make another patch